framework/src/utils/PetscSupport.C

```diff
@@ -354,6 +354,55 @@ petscNonlinearConverged(SNES /*snes*/,
   PetscFunctionReturn(PETSC_SUCCESS);
 }
 
+PetscErrorCode
+petscLinearConverged(
+    KSP /*ksp*/, PetscInt it, PetscReal /*norm*/, KSPConvergedReason * reason, void * ctx)
+{
+  PetscFunctionBegin;
+  FEProblemBase & problem = *static_cast<FEProblemBase *>(ctx);
+
+  // execute objects that may be used in convergence check
+  // TODO: add a dedicated flag once support is fully implemented.
+  // Right now, setting objects to execute on this flag would be ignored except in the
+  // linear-system-only use case.
+  problem.execute(EXEC_NONLINEAR_CONVERGENCE);
+
+  // perform the convergence check
+  Convergence::MooseConvergenceStatus status;
+  // TODO: This API could become about the "solverConvergenceCheck" to apply to both linear and
+  // nonlinear system convergence checks.
+  if (problem.getFailNextNonlinearConvergenceCheck())
```

Comment:
Either rename it `getFailNextConvergenceCheck` (or something) or add new API. I don't want to use conceptually wrong names throughout this function. Same comment applies to all uses of 'nonlinear' in this function

Reply:
The TODO comment should be gone before this is merged

Reply:
gone. I think the APIs can be determined when we enable linear-within-nonlinear convergence checks. It is using dedicated APIs now, with the nonlienar APIs forwarding to the new more general APIs